components/nimbus/src/persistence.rs

```diff
@@ -257,6 +311,113 @@ impl Database {
         self.meta_store
             .put(&mut writer, DB_KEY_DB_VERSION, &DB_VERSION)?;
         writer.commit()?;
+        log::debug!("transaction commited");
+        Ok(())
+    }
+
+    fn clear_all_db_stores_except_updates(
+        &self,
+        writer: &mut rkv::Writer<rkv::backend::SafeModeRwTransaction>,
+    ) -> Result<(), NimbusError> {
+        self.meta_store.clear(writer)?;
+        self.experiment_store.clear(writer)?;
+        self.enrollment_store.clear(writer)?;
+        Ok(())
+    }
```

Comment:
I'd like to point out that clearing the meta_store is wiping the database version, the nimbus_id, and whether or not the user has opted out of experiments globally, which I don't know if it's entirely necessary or desired.

Reply:
Also, it is a design requirement of Nimbus to try and report experiments we have been enrolled in for some time after the enrollment has ended (30 days, I think), and this would potentially wipe away that information.  Currently I don't think that there are any real experiments that have the potential for this to happen in, so maybe it's just more of a "let's don't forget".

Reply:
I'm not as concerned about the historical reporting of experiments as I am about losing the global opt out setting specifically, and less specifically the nimbus_id.  If we dump this database store then could we potentially opt a user back into experiments if they had previously opted out, is my fear.

Rotating the nimbus_id isn't the end of the world, but we should avoid doing it unless we have a need as it could potentially affect the user's enrollment bucketing.

Reply:
Fixed.